### PR TITLE
feat: reduce quant error for qwen

### PR DESCRIPTION
**Pull request by Forpee**
I tried several approached to reduce quant error. The best results came from simply increasing scale and fixing resulting overflow panics.

`SCALE=8`:
```
 INFO   Cosine similarity   : 0.695232
 INFO   MSE                 : 7.248738
 INFO   RMSE                : 2.692348
 INFO   Max absolute error  : 14.965368
 INFO   Mean absolute error : 2.088616
 INFO   Relative MSE        : 0.948853
 INFO   KL divergence       : 4.606529
 INFO   Top-1 agreement     : 0.00%
 INFO   Top-5 agreement     : 20.00%
 INFO   Top-10 agreement    : 20.00%
 INFO   Spearman rank corr  : 0.386522
 INFO   Pearson correlation : 0.388934
```

`SCALE=14`:
```
 INFO   Cosine similarity   : 0.997492
 INFO   MSE                 : 0.070934
 INFO   RMSE                : 0.266335
 INFO   Max absolute error  : 1.320830
 INFO   Mean absolute error : 0.209358
 INFO   Relative MSE        : 0.009285
 INFO   KL divergence       : 0.029175
 INFO   Top-1 agreement     : 100.00%
 INFO   Top-5 agreement     : 100.00%
 INFO   Top-10 agreement    : 90.00%
 INFO   Spearman rank corr  : 0.995235
 INFO   Pearson correlation : 0.995790
 ```

`SCALE=20`: 
```
 INFO   Cosine similarity   : 0.999919
 INFO   MSE                 : 0.002294
 INFO   RMSE                : 0.047892
 INFO   Max absolute error  : 0.235706
 INFO   Mean absolute error : 0.038411
 INFO   Relative MSE        : 0.000300
 INFO   KL divergence       : 0.000482
 INFO   Top-1 agreement     : 100.00%
 INFO   Top-5 agreement     : 100.00%
 INFO   Top-10 agreement    : 100.00%
 INFO   Spearman rank corr  : 0.999855
 INFO   Pearson correlation : 0.999874
 ```

### CODE REVIEW

File: atlas-onnx-tracer/src/ops/cos.rs [hunk 1]
```diff
@@ -6,8 +6,15 @@ use crate::{
...
-        tensor::ops::nonlinearities::cos(&remainder, self.scale.into())
+        #[cfg(feature = "fused-ops")]
+        {
+            tensor::ops::nonlinearities::cos(inputs[0], self.scale.into())
+        }
+        #[cfg(not(feature = "fused-ops"))]
+        {
+            let remainder = tensor::ops::nonlinearities::const_rem(inputs[0], FOUR_PI_APPROX);
+            tensor::ops::nonlinearities::cos(&remainder, self.scale.into())
+        }
```
Comment (AntoineF4C5) [inline atlas-onnx-tracer/src/ops/cos.rs:17]: Are we removing completely the part of taking the remainder in the precision evaluation ?
This remainder operator could be a major source of mismatching between the expected and obtained output, if the values in the output go out of the range of the associated lookup table. Hence I believe it would be best to keep the remainder operator to evaluate the model precision.

**Review by AntoineF4C5 — COMMENTED** [review]
Great job, it is nice to be able to evaluate the model precision and identify worst operators.
I left a comment on evaluating the precision of cos/sin operators.

**Comment by pdezk** [comment]
Possibly relevant to the error hunt: `examples/qwen.rs` feeds `attention_mask` as raw 1s
(second input mislabeled `token_type_ids`), which de-quantizes to 0 — the uniform mask bias
then cancels row-wise, so the example's attention is effectively unmasked. Any
quantization-error measurement routed through that example runs on a non-causal graph.
Details/repro in #260 (found while validating the masking envelope for #258).
